src/libraries/Common/src/System/Security/Cryptography/ECCng.ImportExport.cs

```diff
@@ -495,8 +495,8 @@ private static ECCurve.ECCurveType ConvertToCurveTypeEnum(Interop.BCrypt.ECC_CUR
                 }
                 finally
                 {
-                    Marshal.FreeHGlobal(descPtr);
-                    Marshal.FreeHGlobal(buffPtr);
+                    NativeMemory.Free((void*)(nint)descPtr);
```

Comment:
It would be nice to change the type of the variables to `void*` or other appropriate pointer type to make these back and forth casts unnecessary. These casts do not look good.